modules/tensor_mechanics/src/userobjects/CrackFrontPointsProvider.C

```diff
@@ -0,0 +1,22 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "CrackFrontPointsProvider.h"
+
+template <>
+InputParameters
+validParams<CrackFrontPointsProvider>()
+{
+  InputParameters params = validParams<GeneralUserObject>();
+  params.addClassDescription("Crack front points provider base class. Override the "
```

Comment:
Maybe reword this description as:
`Base class for a class that can provide a set of points along a crack front. The virtual functions must be overridden by a derived class to provide this functionality.`